### ISSUE TITLE
Operator Usage Whitespace in String

### ISSUE THREAD

**Issue by khoogheem** [realm/SwiftLint#4663]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

the Operator Usage Whitespace rule triggers In a string. like:
"PAYMENT_PROVIDERS:\(submit ? "POST": "GET")_PAYOUT_REQUEST"

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew

**Comment by SimplyDanny** [comment]
The example is not even a valid string. Could you please provide a proper code snippet that triggers the rule?

**Comment by khoogheem** [comment]
```
    static func payoutRequest(submit: Bool, paymentProviderId: String, withdrawalAmount: Int) -> ServiceRequest {
   
       ...
   
        let logMessage = { () -> String in
            "PAYMENT_PROVIDERS:\(submit ? "POST": "GET")_PAYOUT_REQUEST"
        }
        ....
}
```

**Comment by marcelofabri** [comment]
This is expected - you're missing a space after `"POST"`. Note that this is triggering because it's a string interpolation, so it's treated as code.
